### DISCUSSION TITLE
Progression isn't working when use gz compression in the build

### DISCUSSION THREAD

**Discussion by lchabrand** [jeffreylanters/react-unity-webgl#286] [Help Wanted!]
**Is your feature request related to a problem? Please describe.**
I have an unity context like this : 
`const unityContext = new UnityContext({
    loaderUrl: "build/Expo.loader.js",
    dataUrl: "build/Expo.data.gz",
    frameworkUrl: "build/Expo.framework.js.gz",
    codeUrl: "build/Expo.wasm.gz",
});`
with the .gz compression.
The progression listener isn't working properly, staying to 0, then go to 1 when it's ready.
It's working well without the .gz compression on the build.

**Describe the solution you'd like**
Make the progression listener compatible with gz compression.

**Comment by jeffreylanters** [comment]
Hi. Compression has nothing to do with this module, you have to configure this on your server.

**Comment by lchabrand** [comment]
Hi, thanks for your answer.
Yes i know, and my server is properly configured and everything is working well, except the progress listener which isn't incrementing.
It goes from 0 to 100 in maybe 1 minute (depending of your connection) but working well without .gz extension in my `unityContext`.

**Reply by manzinge** [reply]
Hey :)
Unity fixed the compression loading problem you described in the Release of Unity 2021.2.0 (at least for Brotli Compression).
Link: https://unity3d.com/unity/whats-new/2021.2.0
"WebGL: Fixed incorrect loading progress shown in Player when Brotli build compression is used."

So you might think about switching to Brotli instead of Gzip (if that is possible for your usecase).
I hope this somewhat helps you!

**Reply by angela318** [reply]
@Ichabrand, did you get the gz extension working? I am also trying to make it work here.

**Reply by ShawnCentra** [reply]
Also couldnt get .gz to work. 

I heard we might have to add support for it on our end (server) 

https://github.com/jeffreylanters/react-unity-webgl/issues/126#issuecomment-716598950 

found this post that goes into some detail.
